cuqi/experimental/mcmc/_gibbs.py

```diff
@@ -0,0 +1,217 @@
+from cuqi.distribution import JointDistribution
+from cuqi.experimental.mcmc import SamplerNew
+from cuqi.samples import Samples
+from typing import Dict
+import numpy as np
+import warnings
+
+try:
+    from progressbar import progressbar
+except ImportError:
+    def progressbar(iterable, **kwargs):
+        warnings.warn("Module mcmc: Progressbar not found. Install progressbar2 to get sampling progress.")
+        return iterable
+
+# Not subclassed from SamplerNew as Gibbs handles multiple samplers and samples multiple parameters
+# Similar approach as for JointDistribution
+class GibbsNew: 
+    """
+    Gibbs sampler for sampling a joint distribution.
+
+    Gibbs sampling samples the variables of the distribution sequentially,
+    one variable at a time. When a variable represents a random vector, the
+    whole vector is sampled simultaneously.
+    
+    The sampling of each variable is done by sampling from the conditional
+    distribution of that variable given the values of the other variables.
+    This is often a very efficient way of sampling from a joint distribution
+    if the conditional distributions are easy to sample from.
+
+    In each Gibbs step, the corresponding sampler has the initial_point set
+    to the previous sample and the sampler is reinitialized.
+
+    Parameters
+    ----------
+    target : cuqi.distribution.JointDistribution
+        Target distribution to sample from.
+    
+    sampling_strategy : dict
+        Dictionary of sampling strategies for each parameter.
+        Keys are parameter names.
+        Values are sampler objects.
+
+    Example
+    -------
+    .. code-block:: python
+
+        import cuqi
+        import numpy as np
+
+        # Model and data
+        A, y_obs, probinfo = cuqi.testproblem.Deconvolution1D(phantom='square').get_components()
+        n = A.domain_dim
+
+        # Define distributions
+        d = cuqi.distribution.Gamma(1, 1e-4)
+        l = cuqi.distribution.Gamma(1, 1e-4)
+        x = cuqi.distribution.GMRF(np.zeros(n), lambda d: d)
+        y = cuqi.distribution.Gaussian(A, lambda l: 1/l)
+
+        # Combine into a joint distribution and create posterior
+        joint = cuqi.distribution.JointDistribution(d, l, x, y)
+        posterior = joint(y=y_obs)
+
+        # Define sampling strategy
+        sampling_strategy = {
+            'x': cuqi.experimental.mcmc.LinearRTONew(maxit=15),
```

Comment:
We discussed allowing the specification of how many samples to run for individual samplers. Not sure if the `maxit` here does exactly that, and if so if that is already implemented? In any case, we agreed to include in this PR the functionality to specify a number of samples to run, I believe simplify by looping over `step` the provided nunber of samples. I suppose it is to be determined how the user would provide the number of samples to run each sampler, perhaps that is done exactly like in this line.

Reply:
Added parameter `sampling_steps`

Reply:
Nice! See new comment on how to use this. Perhaps we can discuss the interface for this at stand up tomorrow.

Reply:
Will use Amals comment to continue discussion.